proxy/src/control/destination/background.rs

```diff
@@ -207,17 +208,14 @@ where
                                 addrs: Exists::Unknown,
                                 query,
                                 dns_query: None,
+                                dns_resolver: self.dns_resolver.clone(),
```

Comment:
I think i'm overlooking something. Why is the concrete benefit of cloning `dns_resolver` into each  `DestinationSet`? Does this `clone()` save other `clone()`s later, or is there something else I'm missing?

Reply:
@briansmith 
> Does this `clone()` save other `clone()`s later

Yeah, that's correct. By cloning it once per `DestinationSet`, we no longer clone it every time we call `DestinationSet::reset_dns_query` (which I would expect to happen much more frequently than `DestinationSet::new`).

Reply:
> Yeah, that's correct. By cloning it once per DestinationSet, we no longer clone it every time we call DestinationSet::reset_dns_query (which I would expect to happen much more frequently than DestinationSet::new).

Isn't that accomplished independently by the change from `fn lookup_ip(self, name: &Name)` to `fn lookup_ip(&self, name: &Name)`?

Reply:
> Isn't that accomplished independently by the change from `fn lookup_ip(self, name: &Name)` to `fn lookup_ip(&self, name: &Name)`?

...yes, it is. I completely overlooked that. Oops. 😓 

I'll undo this part of the change and replace it with just removing the clones in `reset_dns_query`. 

Thanks for catching my mistake here!

Reply:
@briansmith I've edited this PR to remove that part of the change. The rest should be reviewable now.